addons/account/wizard/account_resequence.py

```diff
@@ -0,0 +1,160 @@
+            raise UserError(_('You can only resequence items from the same journal'))
+        values['move_ids'] = [(6, 0, active_move_ids.ids)]
+        return values
+
+    @api.depends('first_name')
+    def _compute_sequence_number_reset(self):
+        for record in self:
+            record.sequence_number_reset = self.env['account.move']._deduce_sequence_number_reset(record.first_name)
+
+    @api.depends('move_ids')
+    def _compute_first_name(self):
+        self.first_name = ""
+        for record in self:
+            if record.move_ids:
+                record.first_name = min(record.move_ids._origin.mapped('name'))
+
+    @api.depends('new_values', 'ordering')
+    def _compute_preview_moves(self):
+        """Reduce the computed new_values to a smaller set to display in the preview."""
+        for record in self:
+            new_values = sorted(json.loads(record.new_values).values(), key=lambda x: x['server-date'], reverse=True)
+            changeLines = []
+            in_elipsis = 0
+            previous_line = None
+            for i, line in enumerate(new_values):
+                if i < 3 or i == len(new_values) - 1 or line['new_by_name'] != line['new_by_date'] \
+                 or (self.sequence_number_reset == 'year' and line['server-date'][0:4] != previous_line['server-date'][0:4])\
+                 or (self.sequence_number_reset == 'month' and line['server-date'][0:7] != previous_line['server-date'][0:7]):
+                    if in_elipsis:
+                        changeLines.append({'current_name': '... (%s other)' % str(in_elipsis), 'new_by_name': '...', 'new_by_date': '...', 'date': '...'})
+                        in_elipsis = 0
+                    changeLines.append(line)
+                else:
+                    in_elipsis += 1
+                previous_line = line
+
+            record.preview_moves = json.dumps({
+                'ordering': record.ordering,
+                'changeLines': changeLines,
+            })
+
+    @api.depends('first_name', 'move_ids', 'sequence_number_reset')
+    def _compute_new_values(self):
+        """Compute the proposed new values.
+
+        Sets a json string on new_values representing a dictionary thats maps account.move
+        ids to a disctionay containing the name if we execute the action, and information
+        relative to the preview widget.
+        """
+        def _get_move_key(move_id):
+            if self.sequence_number_reset == 'year':
+                return move_id.date.year
+            elif self.sequence_number_reset == 'month':
+                return (move_id.date.year, move_id.date.month)
+            return 'default'
+
+        def _sort_by_name_key(name):
+            match = re.match(self.env['account.move']._sequence_fixed_regex, name)
+            return (match.group('prefix1'), int(match.group('seq') or '0'), match.group('suffix'))
+
+        self.new_values = "{}"
+        for record in self.filtered('first_name'):
```

Comment:
whaaaat?

Reply:
To avoid one level of indentation with
```python3
            for record in self:
                if record.first_name:  # this one
                ...
                    for j, grouped_list in enumerate(sorted.values()):
                        for move in grouped_list:
                            ...
```